ddtrace/internal/utils/http.py

```diff
@@ -176,10 +179,10 @@ def w3c_get_dd_list_member(context):
             and k not in [SAMPLING_DECISION_TRACE_TAG_KEY, USER_ID_KEY]
         ):
             # for key replace ",", "=", and characters outside the ASCII range 0x20 to 0x7E
-            # for value replace ",", ";", ":" and characters outside the ASCII range 0x20 to 0x7E
+            # for value replace ",", ";", "~" and characters outside the ASCII range 0x20 to 0x7E
             next_tag = "{}:{}".format(
-                re.sub("_dd.p.", "t.", re.sub(r",| |=|[^\x20-\x7E]+", "_", k)),
-                re.sub(_W3C_TRACESTATE_INVALID_CHARS_REGEX, "_", v),
+                w3c_encode_tag("_dd.p.", "t.", w3c_encode_tag(r",| |=|[^\x20-\x7E]+", "_", k)),
```

Comment:
rather than w3c_encode_tag twice on the key here, should I change the method to take a dict like {regex: replacement} and loop through it? This is the only instance of two replacements being done on a single str so wasn't sure what was cleaner.